src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
+        confirmation.close();
+        card._handleDeleteCard();
+      })
+      .catch((err) => console.log(err));
+  });
 }
 
-function createCard(cardData) {
-  return new Card(cardData, "#card-template", handleImageClick).getView();
+// Handle Like click
+function handleLikeCard(card) {
+  if (card._isLiked) {
+    api
+      .removeLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(card._id)
+      .then((res) => {
+        card.updateIsLiked(res.isLiked);
+      })
+      .catch((err) => console.log(err));
+  }
 }
+
 /* -------------------------------------------------------------------------- */
 /*                               Event Listeners                              */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Form
+
+// Edit Profile Form
 profileEditBtn.addEventListener("click", () => {
   const userInput = user.getUserInfo();
-  editProfilePopup.setInputValues({
-    title: userInput.name,
-    subheader: userInput.about,
-  });
-  profileEditFormValidator.resetValidation();
+  profileTitleInput.value = userInput.name;
+  profileDescriptionInput.value = userInput.about;
   editProfilePopup.open();
+  profileEditFormValidator.resetValidation();
 });
 
-//New Card Form
+// New Card Form
 addCardBtn.addEventListener("click", () => {
   newCardPopup.open();
   addCardFormValidator.toggleButtonState();
 });
 
+// Delete Card Form
+const confirmation = new PopupDelete({
+  popupSelector: "#delete-card-modal",
+});
+confirmation.setEventListeners();
+
+// Avatar Edit Form
+avatarEditBtn.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
 /* -------------------------------------------------------------------------- */
-/*                               Validation                                 */
+/*                               Validation                                    */
 /* -------------------------------------------------------------------------- */
 const addCardFormValidator = new FormValidator(config, addCardForm);
 addCardFormValidator.enableValidation();
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
 profileEditFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
+
+/* -------------------------------------------------------------------------- */
+/*                           Fetch and Render Data                           */
+/* -------------------------------------------------------------------------- */
+
+Promise.all([api.getUserData(), api.getInitialCards()])
+  .then(([profileData, cards]) => {
+    user.setUserInfo(profileData.name, profileData.about);
+    user.changeAvatar(profileData.avatar);
+    section.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Error fetching data:", err);
```

Comment:
You duplicated the code of initial cards or user info getting. Please, leave `Promise.all` or the 2 separate api calls which you have in lines 54 and 122